webhook/gitlab.go

```diff
@@ -0,0 +1,31 @@
+// Copyright [2022] [Argus]
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package webhook
+
+import (
+	"net/http"
+	"net/url"
+)
+
+// SetGitHubHeaders of the req based on the payload and secret.
+func SetGitLabParameter(req *http.Request, secret string) {
+	q := url.Values{}
+	if req.URL.Query() != nil && len(req.URL.Query()) != 0 {
+		q = req.URL.Query()
+	}
+	q.Add("token", secret)
+	q.Add("ref", "master")
```

Comment:
What if master isn't their leading branch?

Reply:
I have hard-coded the branch because I see no way to set a custom branch over the config.
It is also set to master currently for type GitHub.

For GitLab exists two urls which can be used.
```
https://gitlab.com/api/v4/projects/<PROJECT ID>/trigger/pipeline
https://gitlab.com/api/v4/projects/<PROJECT ID>/ref/<BRANCH>/trigger/pipeline
```

The fisrt needs the Query parameter (current master), the second override the Query parameter branch with the branch in the URL.

Reply:
Alright, I don't see a reason why it wouldn't, but before I merge, have you tested that this works against your own GitLab? (I don't use GitLab)

Reply:
Yes I have tested this with my own GitLab (not gitlab.com).

Reply:
Perfect, thanks for the PR 😄